Doc/c-api/bytes.rst

```diff
@@ -189,6 +189,22 @@ called with a non-bytes parameter.
    to *newpart* (i.e. decrements its reference count).
 
 
+.. c:function:: PyObject* PyBytes_Join(PyObject *sep, PyObject *iterable)
+
+   Similar to ``sep.join(iterable)`` in Python.
+
+   *sep* must be Python :class:`bytes` object, or ``NULL`` which is treated as
+   an empty string.
```

Comment:
"empty string" is misleading, since passing an empty `str` instance is an error.

This is the part where `PyBytes_Join()` differs from `PyUnicode_Join()`. `PyUnicode_Join()` use a single space string separator if the first argument is NULL (although this behavior is not documented). I believe an empty separator is much more useful, but this should be discussed and voted.

Reply:
> "empty string" is misleading, since passing an empty str instance is an error.

Passing an empty string as *sep* is valid. There is an unit test on that. What do you mean?

Reply:
There are tests for empty `bytes` instance, not empty `str` instance. If the latter is accepted, this is a bug.

Reply:
ah, your concern is about Unicode vs bytes empty string. I added a test and I clarified the documentation.